app/test/marionette/lib/views/view.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var Marionette = require('marionette-client'),
+  BUDDY_UP_ORIGIN = require('../buddyup').ORIGIN
+
+function View(client) {
+  this.client = client.scope({
+    searchTimeout: 5000
+  });
+  this.actions = new Marionette.Actions(client);
+}
+module.exports = View;
+
+
+View.prototype = {
+  get pageTitle() {
+    return this.client.findElement(this._SELECTORS.PAGE_TITLE).text();
+  },
+
+  switchToFrame: function() {
+    return _switchToInternalFrame(this.client, this._SELECTORS.MAIN);
+  },
+
+  _SELECTORS: {
+    PAGE_TITLE: 'header h1'
+  }
+};
+
+
+var _switchToInternalFrame = function(client, frameLocator) {
+  return new Promise(function(resolve, reject) {
+    client.switchToFrame(function() {
+      client.apps.switchToApp(BUDDY_UP_ORIGIN);
+      client.findElement(frameLocator, function(err, element) {
+        if (err) {
+          reject(err)
+        } else {
+          client.waitFor(function() {
+            element.displayed();
+            client.switchToFrame(element, resolve);
+          })
```

Comment:
I think the code aboves require some comments to explain what's happening. Like we want to select an iframe sibling and to do that, we need to go to the root node of the OS with `client.switchToFrame` first.

You can save an indentation level with:
```js
if (err) {
  reject(err);
  return;
}

client.waitFor(function() {});
```
This will be nice because the current code suffers from callback hell.